File: app/src/main/java/me/mitul/aij/Helper/HelperLogin.java
package me.mitul.aij.helper;

import android.annotation.SuppressLint;
import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import com.readystatesoftware.sqliteasset.SQLiteAssetHelper;

public class HelperLogin extends SQLiteAssetHelper {
    @SuppressLint("SdCardPath")
    public HelperLogin(Context context) {
        super(context, "AIJ_DB.s3db", "/data/data/me.mitul.aij/databases", null, 1);
        getReadableDatabase().close();
    }

    public int attemptLogin(String email, String pass) {
        String query = "SELECT * FROM MST_User WHERE UserName = '" + email + "' AND Password = '" + pass + "'";
        SQLiteDatabase database = getReadableDatabase();
        Cursor cursor = database.rawQuery(query, null);
        if (cursor.getCount() < 0) return 999;
        if (cursor.moveToFirst()) {
            String userName = cursor.getString(cursor.getColumnIndex("UserName"));
            String password = cursor.getString(cursor.getColumnIndex("Password"));
            return userName.equals(email) && password.equals(pass) ? cursor.getInt(cursor.getColumnIndex("UserID")) : 999;
        }
        cursor.close();
        database.close();
        return 999;
    }

    public void insertUser(String email, String pass) {
        SQLiteDatabase database = getWritableDatabase();
        ContentValues values = new ContentValues();
        values.put("UserName", email);
        values.put("Password", pass);
        database.insert("MST_User", null, values);
        database.close();
    }
}

File: app/src/main/java/me/mitul/aij/helper/HelperLogin.kt
package me.mitul.aij.helper

import android.content.ContentValues
import android.content.Context
import androidx.core.database.getIntOrNull
import androidx.core.database.getStringOrNull
import com.readystatesoftware.sqliteasset.SQLiteAssetHelper
import me.mitul.aij.utils.Consts

class HelperLogin(context: Context?) : SQLiteAssetHelper(
    context, Consts.DB_NAME, Consts.DB_PATH, null, Consts.DB_VERSION
) {
    private object Keys {
        const val DB_TBL_NAME = "User"
        const val DB_COL_USER_ID = "user_id"
        const val DB_COL_USERNAME = "username"
        const val DB_COL_PASSWORD = "password"
    }

    fun login(username: String, password: String): Int {
        val cursor = readableDatabase.query(
            Keys.DB_TBL_NAME, null,
            "${Keys.DB_COL_USERNAME}= ? AND ${Keys.DB_COL_PASSWORD} = ?",
            arrayOf(username, password), null, null, null
        )

        if (cursor.count < 0) return -1
        if (cursor.moveToFirst()) {
            val user = cursor.getStringOrNull(cursor.getColumnIndex(Keys.DB_COL_USERNAME))
            val pass = cursor.getStringOrNull(cursor.getColumnIndex(Keys.DB_COL_PASSWORD))
            if (user != username || pass != password) return -1
            return cursor.getIntOrNull(cursor.getColumnIndex(Keys.DB_COL_USER_ID)) ?: -1
        }
        cursor.close()
        return -1
    }
}
